checks/default.nix

```diff
@@ -14,6 +14,13 @@ let
   x64Linux = pkgs.hostPlatform.system == "x86_64-linux";
 in
 {
+  bzip2Sys = myLib.buildPackage {
```

Comment:
Thanks for cleaning up the test!

What's interesting is I tried commenting out the changes in `installCargoArtifactsHook.sh` but this test still passed without them :thinking: 

(Overall I'm fine to add the `w` flag on all files, but it would be nice if this test was able to reproduce the original issue, otherwise it won't catch any potential regressions)

Reply:
Ah found the issue! Including `nativeBuildInputs = [ pkgs.pkg-config ];` reproduces the issue! Except the "fix" to make things writable before `tar`ing them doesn't seem to work (rather the test keeps failing). Yet setting `strictDeps = true;` _does_ fix things, which leads me to believe something funny is happening when `pkg-config` is being listed as a dependency :thinking:

Reply:
Any change in the list of build inputs can lead to `PKG_CONFIG_PATH` changing, which causes certain build scripts to be rebuild by `cargo`. Withtout `strictDeps= true;` inputs of inputs are included as well, so it's easy to trigger accidental recompilation.

Reply:
Ah hmm yeah.. apparently I forgot to add use-zstd to the test. Fixed that now.

Reply:
Ah thanks for fixing that test, looks like it reproduces the issue now